grakn-test/src/test/java/ai/grakn/test/graql/analytics/AnalyticsTest.java

```diff
@@ -903,6 +903,9 @@ public void testComputingUsingDegreeResource() throws GraknValidationException {
 
         // compute count
         analytics = new Analytics(graph.getKeyspace(), Collections.singleton("degree"), new HashSet<>());
+
+        //Not sure why the above seems to be closing the graph now.
+        graph = Grakn.factory(Grakn.DEFAULT_URI, graph.getKeyspace()).getGraph();
```

Comment:
Analytics closes the graph when it performs a mutation because we may need to change the ontology - set certain types to play a role in the resource relation. If we use the graph we are passed it may have uncommitted changes that were not meant to be persisted.